include/posix/unistd.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef __POSIX_UNISTD_H__
+#define __POSIX_UNISTD_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include_next <unistd.h>
+#include "sys/types.h"
+
+/**
+ * @brief Sleep for a specified number of seconds.
+ *
+ * See IEEE 1003.1
+ */
+static inline int sleep(unsigned int seconds)
+{
+	k_sleep(K_SECONDS(seconds));
+	return 0;
+}
+/**
+ * @brief Suspend execution for microsecond intervals.
+ *
+ * See IEEE 1003.1
+ */
+static inline int usleep(useconds_t useconds)
+{
+	/* Creating Busy wait for time below msecs */
+	k_busy_wait(useconds % USEC_PER_MSEC);
```

Comment:
Remove the busy wait.  Precise timing at the expense of wasted cycles is not the intent of usleep() on any existing platforms I know of.

And if you insist on doing this for precision, then this is the wrong way to do it anyway:  the k_sleep() call will only be precise to within a timer interrupt, which is tick-granular on most systems (i.e. much coarser than a microssecond).  So what you have to do is k_sleep() *first* (probably with some slop subtracted so you don't overshoot) then check the elapsed time with k_cycle_get_32() and busy wait on that until the desired time has expired.

But you don't want to do that.  Just yank the busy wait.

Reply:
hmm I got your idea. 

My intention here was :- 
if time in usec is below 1 msec ==> Create with busy wait.
If it is above or equal to 1 msec ==> create delay with k_sleep

As of now my logic was taking care of first case completely (when usecs are below 1msecs) but it is getting unfair to case where it is greater than 1msec. I will mention this clearly.

I was thinking in another way if delay to give is more than 1 msec shall we take the celling and  create the delay using usec. Something like usecs = 2.5msec then create delay k_sleep(3). This will make sure that thread is not scheduled before time. 
What is your suggestion on this ?